src/components/box/box.pw.tsx

```diff
@@ -0,0 +1,897 @@
+import { expect, test } from "@playwright/experimental-ct-react17";
+import React from "react";
+import type { Page } from "@playwright/test";
+import { getDataElementByValue } from "../../../playwright/components";
+import {
+  assertCssValueIsApproximately,
+  checkAccessibility,
+  getStyle,
+} from "../../../playwright/support/helper";
+import { BoxProps } from "../../../src/components/box";
+import {
+  Default,
+  BoxComponentMulti,
+  BoxShadow,
+  Color,
+  Flex,
+  Gap,
+  Layout,
+  OverflowWrap,
+  Position,
+  Scroll,
+  Spacing,
+} from "../../../src/components/box/components.test-pw";
+
+const colorConstants: [string, string, string][] = [
+  ["red", "rgb(255, 0, 0)", "#FF0000"],
+  ["yellow", "rgb(255, 255, 0)", "#FFFF00"],
+  ["blue", "rgb(0, 0, 255)", "#0000FF"],
+];
+
+const bgConstants: [string, string, string][] = [
+  ["primary", "rgb(0, 125, 90)", "#007D5A"],
+  ["secondary", "rgb(0, 96, 70)", "#006046"],
+  ["tertiary", "rgb(0, 64, 46)", "#00402E"],
+];
+
+const widthConstants: [string, number, number][] = [
+  ["135px", 0.1, 135],
+  ["683px", 0.5, 683],
+  ["1366px", 1, 1366],
+];
+
+const heightConstants: [string, number][] = [
+  ["135px", 135],
+  ["683px", 683],
+  ["1366px", 1366],
+];
+
+const sizeConstants: [string, number, number][] = [
+  ["135px", 0.1, 135],
+  ["683px", 0.5, 683],
+  ["1366px", 1, 1366],
+];
+
+const verifyScrollbarVariant = async (
+  page: Page,
+  thumbColor: string,
+  trackColor: string
+) => {
+  const boxLocator = await getDataElementByValue(page, "boxone");
+  const colorValueThumb = await getStyle(
+    boxLocator,
+    "background-color",
+    "-webkit-scrollbar-thumb"
+  );
+  const colorValueTrack = await getStyle(
+    boxLocator,
+    "background-color",
+    "-webkit-scrollbar-track"
+  );
+  await expect(colorValueThumb).toBe(thumbColor);
+  await expect(colorValueTrack).toBe(trackColor);
+};
+
+test.describe("should render Box component", () => {
+  colorConstants.forEach(([color, rgbValue]) => {
+    test(`should verify text is ${color} when color prop is passed as string`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default color={color} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveAttribute("color", color);
```

Comment:
**question**: Do we need these attribute assertions here and in the other tests where we are checking CSS values, when we also have CSS assertions in these tests as well?

Reply:
Ah good point, double checking the mock components it seems we may be accidentally passing CSS attributes as HTML attributes which is incorrect, so we will need to amend this at a later stage

![Screenshot 2023-09-15 at 14 25 38](https://github.com/Sage/carbon/assets/18368713/54fb21af-1b73-4a0a-a640-71a024220d3b)